From 248976a9a924008897aac6ff6cb22a98eee1d7c7 Mon Sep 17 00:00:00 2001
From: Dylan Hung <dylan_hung@aspeedtech.com>
Date: Wed, 1 Dec 2021 14:14:09 +0800
Subject: [PATCH] usb: aspeed: Hotfix of the cache invalidation

This is a hotfix of the cache invalidation.  The cache controller
detects the change of scua58[1] from b'0 to b'1, then triggers the data
cache invalidation.  So we need to clear this bit before we set the bit.

Signed-off-by: Dylan Hung <dylan_hung@aspeedtech.com>
Change-Id: Icfd2da8a50facb34c83da0f982bdda954bbc6a8d
---
 drivers/usb/device/usb_dc_aspeed.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/device/usb_dc_aspeed.c b/drivers/usb/device/usb_dc_aspeed.c
index 4e30b4d70e18..dd999048dff2 100644
--- a/drivers/usb/device/usb_dc_aspeed.c
+++ b/drivers/usb/device/usb_dc_aspeed.c
@@ -135,8 +135,8 @@ LOG_MODULE_REGISTER(usb_dc_aspeed);
 
 #define SCU_BASE		0x7e6e2000
 #define DCACHE_INVALID() { \
-	sys_write32(sys_read32(SCU_BASE + 0xa58) | 0x2, SCU_BASE + 0xa58); \
 	sys_write32(sys_read32(SCU_BASE + 0xa58) & ~0x2, SCU_BASE + 0xa58); \
+	sys_write32(sys_read32(SCU_BASE + 0xa58) | 0x2, SCU_BASE + 0xa58); \
 }
 
 enum ep_state {
-- 
2.25.1

